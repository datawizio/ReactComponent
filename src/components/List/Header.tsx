import React from "react";

import { List } from "antd";

import { ListItemHeaderProps } from "./types.d";

const ListItemHeader: React.FC<ListItemHeaderProps> = ({ title }) => {
  return <List.Item className="list-item-header">{title}</List.Item>;
};

export default ListItemHeader;
