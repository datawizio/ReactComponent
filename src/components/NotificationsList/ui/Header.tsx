import React from "react";

import Checkbox from "../../Checkbox";
import { Button } from "antd";

import { useTranslation } from "react-i18next";
import { CheckboxChangeEvent } from "antd/es/checkbox";

import "./Header.less";

interface ListHeaderProps {
  checkedCount: number;
  checkboxStatus: null | "partial" | "all";
  total?: number;
  actions?: any;
  onChecked: (e: CheckboxChangeEvent) => void;
  onCheckAll: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export const ListHeader: React.FC<ListHeaderProps> = ({
  checkboxStatus,
  checkedCount,
  total,
  actions,
  onChecked,
  onCheckAll
}) => {
  const { t } = useTranslation();

  return (
    <div className="dw-list-header">
      <div>
        <Checkbox
          checked={checkboxStatus === "all"}
          indeterminate={checkboxStatus === "partial"}
          onChange={onChecked}
        />
      </div>
      {checkedCount > 0 && (
        <div className="selected-count">
          {t("SELECTED_NOTIFICATIONS", { count: checkedCount, total })}
        </div>
      )}
      {checkboxStatus === "all" && (
        <div className="select-all-container">
          <Button type="link" onClick={onCheckAll}>
            {t("SELECT_ALL_NOTIFICATIONS", { total })}
          </Button>
        </div>
      )}
      <div className="fill-place"></div>
      {actions}
    </div>
  );
};
