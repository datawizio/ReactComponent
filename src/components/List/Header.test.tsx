import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import Header from "./Header";

const mockProps = {
  title: "title"
};

const setUp = (props?) => mount(<Header {...props} />);

describe("ListHeader component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("ListHeader render  with correct title", () => {
    const item = component.find(".list-item-header");
    expect(item.first().html()).toContain(mockProps.title);
  });
});
