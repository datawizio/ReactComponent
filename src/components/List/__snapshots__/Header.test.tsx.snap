// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader component ListHeader rendered correctly 1`] = `
<ListItemHeader
  title="title"
>
  <Item
    className="list-item-header"
  >
    <li
      className="ant-list-item list-item-header"
    >
      title
    </li>
  </Item>
</ListItemHeader>
`;
